assets/src/hooks/useSearchResultsByProperty.ts

```diff
@@ -0,0 +1,82 @@
+import { Socket } from "phoenix"
+import { PropertyLimits } from "../models/searchQuery"
+import { Loading, Ok } from "../util/fetchResult"
+import {
+  LimitedSearchResults,
+  useLimitedSearchResults,
+} from "./useSearchResults"
+import { useLocationSearchResults } from "./useLocationSearchResults"
+import { LocationSearchResult } from "../models/locationSearchResult"
+import { Vehicle, Ghost } from "../realtime"
+
+export type VehicleResultType =
+  | Ok<LimitedSearchResults<Vehicle | Ghost>>
+  | Loading
+  | null
+
+type LocationResultsType =
+  | Ok<LimitedSearchResults<LocationSearchResult>>
+  | Loading
+  | null
+
+export type SearchResultsByProperty = {
+  vehicle: VehicleResultType
+  operator: VehicleResultType
+  run: VehicleResultType
+  location: LocationResultsType
+}
+
+const useSearchResultsByProperty = (
+  socket: Socket | undefined,
+  text: string,
+  properties: PropertyLimits
+): SearchResultsByProperty => {
```

Comment:
question: do you think it'd be useful to rewrite [`useAutocompleteResults`](https://github.com/mbta/skate/blob/kf/asn/search-autocomplete/assets/src/hooks/useAutocompleteResults.ts) using this function? Seems like we ended up doing the same thing heh!

Reply:
No worries!
and a cleanup ticket afterward sounds good to me 👍🏻

Reply:
One other thing to consider is that the functionality for location autocomplete vs. results will actually be different (though that can probably be handled by an extra argument to the hook specifying which one you want).